src/com/projectkorra/projectkorra/earthbending/passive/DensityShift.java

```diff
@@ -37,6 +37,8 @@ public static boolean softenLanding(final Player player) {
 		} else if (bPlayer.canMetalbend() && ElementalAbility.isMetalBlock(block)) {
 			return true;
 		}
+		if ((player.getLocation().getY() % 1) != 0)
```

Comment:
This check should never run? Dividing by 1 never has a remainder, right??

Reply:
that is used to check if their Y is a whole number or not. If it’s not, which it isn’t when landing on grass paths or farmland blocks, then that statement will return true and thus my code will attempt to teleport a bit above the block to prevent falling into it.

Reply:
It also isn't a whole number in all of the following instances [https://minecraft.gamepedia.com/Solid_block#Height]

And given the code, your stuff will teleport them up before it checks whether there is an earthbendable block underneath them. So lets say someone lands on a ladder. It'll teleport them up exactly .1 blocks.

Reply:
it can be moved, eZ fix